editor/src/components/canvas/canvas-strategies/relative-move-strategy.spec.browser2.tsx

```diff
@@ -0,0 +1,393 @@
+import React from 'react'
+import { createBuiltInDependenciesList } from '../../../core/es-modules/package-manager/built-in-dependencies-list'
+import { elementPath } from '../../../core/shared/element-path'
+import {
+  ElementInstanceMetadata,
+  ElementInstanceMetadataMap,
+  SpecialSizeMeasurements,
+} from '../../../core/shared/element-template'
+import {
+  canvasPoint,
+  canvasRectangle,
+  CanvasVector,
+  localRectangle,
+} from '../../../core/shared/math-utils'
+import { ElementPath } from '../../../core/shared/project-file-types'
+import { emptyModifiers } from '../../../utils/modifiers'
+import { EditorState } from '../../editor/store/editor-state'
+import { foldAndApplyCommands } from '../commands/commands'
+import {
+  getEditorState,
+  makeTestProjectCodeWithSnippet,
+  testPrintCodeFromEditorState,
+} from '../ui-jsx.test-utils'
+import { absoluteMoveStrategy } from './absolute-move-strategy'
+import { pickCanvasStateFromEditorState } from './canvas-strategies'
+import { defaultCustomStrategyState } from './canvas-strategy-types'
+import { InteractionSession, StrategyState } from './interaction-state'
+import { createMouseInteractionForTests } from './interaction-state.test-utils'
+
+const prepareEditorState = (
+  codeSnippet: string,
+  selectedViews: Array<ElementPath>,
+): EditorState => {
+  return {
+    ...getEditorState(makeTestProjectCodeWithSnippet(codeSnippet)),
+    selectedViews: selectedViews,
+  }
+}
+
+const defaultMetadata: ElementInstanceMetadataMap = {
+  'scene-aaa': {
+    elementPath: elementPath([['scene-aaa']]),
+  } as ElementInstanceMetadata,
+  'scene-aaa/app-entity': {
+    elementPath: elementPath([['scene-aaa', 'app-entity']]),
+  } as ElementInstanceMetadata,
+  'scene-aaa/app-entity:aaa': {
+    elementPath: elementPath([['scene-aaa', 'app-entity'], ['aaa']]),
+  } as ElementInstanceMetadata,
+  'scene-aaa/app-entity:aaa/bbb': {
+    elementPath: elementPath([
+      ['scene-aaa', 'app-entity'],
+      ['aaa', 'bbb'],
+    ]),
+    specialSizeMeasurements: {
+      immediateParentBounds: canvasRectangle({ x: 0, y: 0, width: 400, height: 400 }),
+      coordinateSystemBounds: canvasRectangle({ x: 0, y: 0, width: 400, height: 400 }),
+    } as SpecialSizeMeasurements,
+    globalFrame: canvasRectangle({ x: 50, y: 50, width: 250, height: 300 }),
+    localFrame: localRectangle({ x: 50, y: 50, width: 250, height: 300 }),
+  } as ElementInstanceMetadata,
+}
+
+function dragByPixels(
```

Comment:
I think rather than using this pattern which creates and updates the `InteractionSession`s directly, we would be better off to use the pattern where we fire mouse events. e.g. we in the absolute move tests we fire one to select the element, and then another to drag it here: https://github.com/concrete-utopia/utopia/blob/master/editor/src/components/canvas/canvas-strategies/absolute-move-strategy.spec.browser2.tsx#L50-L58

Reply:
https://github.com/concrete-utopia/utopia/pull/2604/commits/a22bbd3f6e432073143d829bd39db8fd64a93f96 <- should this be acceptable? if so it should also address @balazsbajorics comment above